gson/src/main/java/com/google/gson/GsonBuilder.java

```diff
@@ -275,6 +278,18 @@ public GsonBuilder enableComplexMapKeySerialization() {
     return this;
   }
 
+  /**
+   * Configure Gson to deserialize duplicate map key. By default, Gson throw a {@code JsonSyntaxException} when there are more than
+   * one same key.
+   *
```

Comment:
Might be good to add a security note:
```suggestion
   *
   * <p>Note that enabling support for duplicate maps keys is discouraged because it can make an application less secure.
   * When an application interacts with other components using different JSON libraries, they might treat duplicate keys
   * differently, allowing an attacker to circumvent security checks.
   *
```